xbmc/video/VideoDatabase.h

```diff
@@ -180,6 +184,40 @@ enum class VideoDbContentType
   MUSICALBUMS = 6
 };
 
+template<>
+struct fmt::formatter<VideoDbContentType> : fmt::formatter<std::string_view>
+{
+  template<typename FormatContext>
+  constexpr auto format(const VideoDbContentType& type, FormatContext& ctx)
+  {
+    return fmt::formatter<std::string_view>::format(enumToSV(type), ctx);
+  }
+
+private:
+  static constexpr std::string_view enumToSV(VideoDbContentType type)
+  {
+    using namespace std::literals::string_view_literals;
+    switch (type)
+    {
+      case VideoDbContentType::UNKNOWN:
+        return "unknown"sv;
+      case VideoDbContentType::MOVIES:
+        return "movies"sv;
+      case VideoDbContentType::TVSHOWS:
+        return "TV shows"sv;
+      case VideoDbContentType::MUSICVIDEOS:
+        return "music videos"sv;
+      case VideoDbContentType::EPISODES:
+        return "episodes"sv;
+      case VideoDbContentType::MOVIE_SETS:
+        return "movie sets"sv;
+      case VideoDbContentType::MUSICALBUMS:
+        return "music albums"sv;
+    };
+    throw std::invalid_argument("no videodb content string found");
```

Comment:
could be just me, isn't throwing excessive?
As far as I know Kodi only has a catch at the top level of threads except for few specific areas of the code (db).
```
default:
  return {};
```
 would be fine?

Reply:
I think an exception is OK here. Adding a `default` has the downside that the compiler stops issuing warnings if not all enum cases are handled. Also, if we should ever get something that's not a `VideoDbContentType` something went really wrong and an exception is adequate.